sorc/apply_incr_noahmp_snow.f90

```diff
@@ -319,19 +339,178 @@ subroutine get_fv3_mapping(myrank, ens_mem, tile_num, rst_path, date_str, hour_s
+
+    ierr=nf90_open(trim(filename),nf90_nowrite,ncid)
+    call netcdf_err(ierr, 'opening file: '//trim(filename) )
+
+    ! CHECK DIMENSIONS
+    ierr=nf90_inq_dimid(ncid, 'lon', id_dim)
+    call netcdf_err(ierr, 'reading lon id from '//trim(filename) )
+    ierr=nf90_inquire_dimension(ncid,id_dim,len=fres)
+    call netcdf_err(ierr, 'reading lon from '//trim(filename) )
+
+    if ( fres /= res) then
+       print*,'fatal error: dimensions wrong in file '//trim(filename)
+       call mpi_abort(mpi_comm_world, ierr)
+    endif
+
+    ! READ land frac 
+    ierr=nf90_inq_varid(ncid, "land_frac", id_var)
+    call netcdf_err(ierr, 'reading land_frac id' )
+    ierr=nf90_get_var(ncid, id_var, land_frac)
+    call netcdf_err(ierr, 'reading land_frac' )
+
+    ! close file 
+    ierr=nf90_close(ncid)
+    call netcdf_err(ierr, 'closing file: '//trim(filename) )
+    
+    ! Use vtype to exclude glaciers
+    ! OPEN FILE
+    write(rankch, '(i1.1)') (tile_num)
+    restart_file = trim(rst_path)//"/"//date_str//"."//hour_str//"0000.sfc_data.tile"//rankch//".nc"
+
+    inquire(file=trim(restart_file), exist=file_exists)
 
+    if (.not. file_exists) then
+            print *, 'restart_file does not exist, ', &
+                    trim(restart_file) , ' exiting'
+            call mpi_abort(mpi_comm_world, 10) 
     endif
+
+    ierr=nf90_open(trim(restart_file),nf90_write,ncid)
+    call netcdf_err(ierr, 'opening file: '//trim(restart_file) )
  
+    ! REMOVE GLACIER GRID POINTS
+    ierr=nf90_inq_varid(ncid, "vtype", id_var)
+    call netcdf_err(ierr, 'reading vtype id' )
+    ierr=nf90_get_var(ncid, id_var, vtype)
+    call netcdf_err(ierr, 'reading vtype' )
+
+    if (fice_grid) then    
+      ierr=nf90_inq_varid(ncid, "fice", id_var)
+      call netcdf_err(ierr, 'reading fice id' )
+      ierr=nf90_get_var(ncid, id_var, fice)
+      call netcdf_err(ierr, 'reading fice' )   
+    endif
+
+    ! close file
+    ierr=nf90_close(ncid)
+    call netcdf_err(ierr, 'closing file: '//trim(restart_file) )
+
+    slmsk_lfrac = 0
+    do i = 1, res
+        do j = 1, res
+            if ( land_frac(i,j) >= lfrac_thold ) slmsk_lfrac(i,j) = 1
+        enddo
+    enddo
+
+    ! remove land grid cells if ice is present
+    if (fice_grid) then
+        write (6, *) 'ammending mask to exclude sea ice from', trim(restart_file)
+        do i = 1, res
+            do j = 1, res
+                if (fice(i,j) > fice_fhold ) slmsk_lfrac(i,j) = 0
+            enddo
+        enddo
+    endif
+
+    ! remove land grid cells if glacier land type
+    do i = 1, res
+        do j = 1, res
+            if ( vtype(i,j) ==  vtype_landice) slmsk_lfrac(i,j) = 0 ! vtype is integer, but stored as double
+        enddo
+    enddo
+
     ! get number of land points
     len_land_vec = 0
     do i = 1, res 
         do j = 1, res 
-             if ( slmsk(i,j) == 1)  len_land_vec = len_land_vec+ 1  
+             if ( slmsk_lfrac(i,j) == 1)  len_land_vec = len_land_vec+ 1  
```

Comment:
while we're here, maybe add a check that exits if the land mask in the restart (say from vegtype) is not equal to that from the land_frac in the orog file.